### PR TITLE
CI: initial fuzzer jobs

### PR DESCRIPTION
**Pull request by ordian**
See https://github.com/paritytech/ci_cd/issues/64.

The initial version doesn't have automatic notification on failure, but that could be added later. I've spent too much on it already.

### CODE REVIEW

**Comment by lovelaced** [comment]
This looks fine to me, though if you want to go ahead and add a matrix message step it would look something like:

```
  - name: Fuzzer found something
    if: ${{ failure() }}
     uses: s3krit/matrix-message-action@v0.0.3
     with:
         room_id: ${{ secrets.MATRIX_ROOM_ID_FUZZER }}
         access_token: ${{ secrets.MATRIX_ACCESS_TOKEN_FUZZER }}
         message: "uh oh... ${{ github.event.workflow_run.url }}"
         server: "matrix.parity.io"
```

File: .github/workflows/honggfuzz.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: Run nightly fuzzer jobs
+
+on:
+  schedule:
+    - cron: '0 12 * * *'
```
Comment (lovelaced) [inline .github/workflows/honggfuzz.yml:5]: this should probably be:

0 0 * * *

**Review by lovelaced — CHANGES_REQUESTED** [review]

File: .github/workflows/honggfuzz.yml (hunk 1 above)
Comment (ordian) [reply .github/workflows/honggfuzz.yml:5]: good catch, this is a copy-paste error from my own crontab
```suggestion
    - cron: '0 0 * * *'
```

**Review by lovelaced — APPROVED** [review]

**Comment by TriplEight** [comment]
Saw [something](https://github.com/paritytech/polkadot/runs/2103044079#step:8:5) that shouldn't act like this.

**Comment by TriplEight** [comment]
Also, the [default timeout](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) is 360 min.

**Comment by ordian** [comment]
> Saw [something](https://github.com/paritytech/polkadot/runs/2103044079#step:8:5) that shouldn't act like this.

Thanks, fixed in 86333c5.

> Also, the [default timeout](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) is 360 min.

This is 6 hours, we run fuzzers for 5 hours, 1 extra hour should be enough for builds.

**Comment by TriplEight** [comment]
I meant, in general. Are we good running fuzzing on rather weak public github hosts?

**Comment by ordian** [comment]
> I meant, in general. Are we good running fuzzing on rather weak public github hosts?

Well, we want something working now. So I'd say it's good as is for now until we have more CI dev-time to borrow.

**Comment by TriplEight** [comment]
Everything looks good. Maybe give them a debug run before the merge?

**Review by TriplEight — APPROVED** [review]
:+1:

**Pull request merged**